### DISCUSSION TITLE
IMU/Gyro required?

### DISCUSSION THREAD

**Discussion by SpiffyJif** [BroncBotz3481/YAGSL-Example#80] [Q&A] [answered]
My team temporarily doesn’t have a gyro on our robot, but we wanted to get at least a robot-centric teleop running with YAGSL. I don't see a "none" option for the [IMU on the JSON Docs](https://github.com/BroncBotz3481/YAGSL-Example/blob/main/docs/json/devicejson.md#fields). Will copying this example get a robotcentric teleop working without an IMU connected?

**Comment by thenetworkgrinch** [comment, marked as answer]
No, you need an IMU to use YAGSL and i think swerve in general. You can also use the navx, adis, or pigeon.

**Reply by RobertTheBlair** [reply]
An IMU is not **required** for Swerve as a drivetrain.   It is if you want to have the robot maintain a consistent heading automatically, or other field oriented logic.
